pkg/bindings/images/build.go

```diff
@@ -682,7 +682,11 @@ func nTar(excludes []string, sources ...string) (io.ReadCloser, error) {
 						}
 					}
 				}
+
 				name := filepath.ToSlash(strings.TrimPrefix(path, s+separator))
+				if i > 0 {
+					name = path
```

Comment:
This path isn't going to have the benefit of the `ToSlash()` change, and I think we always want to have passed the path through that.

Reply:
... and if `path` is a directory, this replacement will happen multiple times while we walk through it, which sounds wrong.